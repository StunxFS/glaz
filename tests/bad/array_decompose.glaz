fn simple(_a: i32, _b: ...i32) {}

fn main() {
    let x = vec[i32]{ 1, 2 };
    simple(1, 2, ...x);
    simple(1, ...5);
}
