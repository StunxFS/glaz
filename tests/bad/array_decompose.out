tests/bad/array_decompose.glaz:5:6: error: too many arguments to function `simple`
     | 
   5 |     simple(1, 2, ...x);
     |     ^^^^^^^^^^^^^^^^^^
tests/bad/array_decompose.glaz:6:19: error: decomposition can only be used on arrays
     | 
   6 |     simple(1, ...5);
     |                  ^
glaz error: could not compile `array_decompose`, aborting due to 2 previous errors.
